api/v1/types.go

```diff
@@ -7,48 +7,121 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type StateType string
+
+const (
+	StateTypeHealthy      StateType = "Healthy"
+	StateTypeUnhealthy    StateType = "Unhealthy"
+	StateTypeInitializing StateType = "Initializing"
+	StateTypeDegraded     StateType = "Degraded"
+)
+
 type State struct {
-	Name      string            `json:"name,omitempty"`
-	Healthy   bool              `json:"healthy,omitempty"`
-	Health    string            `json:"health,omitempty"`     // Healthy, Degraded, Unhealthy
-	Reason    string            `json:"reason,omitempty"`     // a detailed and processed reason on why the component is not healthy
-	Error     string            `json:"error,omitempty"`      // the unprocessed error returned from the component
-	ExtraInfo map[string]string `json:"extra_info,omitempty"` // any extra information the component may want to expose
+	// Component represents which component generated the state.
+	Component string `json:"component,omitempty"`
+
+	// Name is the name of the state.
+	Name string `json:"name,omitempty"`
+
+	// Health represents the health level of the state,
+	// including StateHealthy, StateUnhealthy and StateDegraded.
+	// StateDegraded is similar to Unhealthy which also can trigger alerts
+	// for users or operators, but what StateDegraded means is that the
+	// issue detected does not affect users’ workload.
+	Health StateType `json:"health,omitempty"`
+
+	// Reason represents what happened or detected by GPUd if it isn’t healthy.
+	Reason string `json:"reason,omitempty"`
+
+	// Error represents the detailed error information, which will be shown
+	//as More Information to help analyze why it isn’t healthy.
+	Error string `json:"error,omitempty"`
 
+	// SuggestedActions represents the suggested actions to mitigate the issue.
 	SuggestedActions *SuggestedActions `json:"suggested_actions,omitempty"`
+
+	// TO BE DEPRECATED
+	DeprecatedHealthy   bool              `json:"healthy,omitempty"`
+	DeprecatedExtraInfo map[string]string `json:"extra_info,omitempty"`
```

Comment:
Shall we remove them?

Reply:
Still being used everywhere for `"data"` field :)

Let me migrate those first